var/spack/repos/builtin/packages/uchardet/package.py

```diff
@@ -27,3 +30,12 @@ def url_for_version(self, version):
         else:
             url = "https://github.com/BYVoid/uchardet/archive/v0.0.5.tar.gz"
         return url
+
+    @when('platform=darwin')
+    def cmake_args(self):
+        args = []
+        # From https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/uchardet.rb
+        args.append('-DCMAKE_INSTALL_NAME_DIR={0}'.format(self.prefix.lib))
+        # From https://github.com/mutationpp/Mutationpp/issues/26
+        args.append('-DCMAKE_MACOSX_RPATH=ON')
+        return args
```

Comment:
I'd avoid the `when` decorator and use `CMakePackage` convenience functions. Something similar to:
```suggestion
    def cmake_args(self):
        args = []
        if self.spec.satisfies('platform=darwin'):
            args += [
                # From https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/uchardet.rb
                self.define('CMAKE_INSTALL_NAME_DIR', self.prefix.lib),
                # From https://github.com/mutationpp/Mutationpp/issues/26
                self.define('CMAKE_MACOSX_RPATH', 'ON')
            ]
        return args
```
(beware I didn't try the code above, so there might be typos).